core/application/utils/geohash.ts

```diff
@@ -16,3 +16,60 @@ export function decodeGeohash(geohash: string): { latitude: number; longitude: n
     return null
   }
 }
+
+function getNeighbors(geohash: string): string[] {
+  return ngeohash.neighbors(geohash)
+}
+
+export function getGeohashNthNeighbors(geohash: string, neighborLevel: number): string[] {
+  const visited = new Set()
+  visited.add(geohash)
+
+  let currentLevel = [{
+    hash: geohash,
+    level: 0
+  }]
+  const nthLevelNeighbors = []
+
+  for (let i = 0; i < neighborLevel; i++) {
+    const nextLevel = []
+
+    for (const current of currentLevel) {
+      const neighbors = getNeighbors(current.hash)
+
+      for (const neighbor of neighbors) {
+        if (visited.has(neighbor)) continue
+
+        visited.add(neighbor)
+        nextLevel.push({
+          hash: neighbor,
+          level: current.level + 1
+        })
+
+        if (current.level + 1 === neighborLevel) {
+          nthLevelNeighbors.push(neighbor)
+        }
+      }
+    }
+    currentLevel = nextLevel
+  }
+
+  return nthLevelNeighbors
+}
+
+export function getDistanceBetweenGeohashes(geohash1: string, geohash2: string): number {
+  const { latitude: lat1, longitude: lon1 } = ngeohash.decode(geohash1)
+  const { latitude: lat2, longitude: lon2 } = ngeohash.decode(geohash2)
+
+  const toRadians = (degrees: number): number => (degrees * Math.PI) / 180
+
+  const R = 6371
```

Comment:
this should be in no magic numbers?

Reply:
Updated in e2ee829f5e60ace41f903a8affb7b9547304e5ee